doc/arch/adr-002-geoprocessing-caching.md

```diff
@@ -0,0 +1,103 @@
+# 002 - Geoprocessing Caching
+
+## Context
+
+Geoprocessing calls for large shapes can take a long time to complete. While users can draw custom shapes, they can also pick from a list of predefined shapes in the system. Geoprocessing the same shape over and over when multiple users select it is wasteful and unnecessary. By caching the geoprocessing results of predefined shapes, or Well Known Areas of Interest (WKAoIs), we can improve user experience and application performance.
```

Comment:
That's a good idea. I tried to think of a quick way of doing it, but after dismissing a bash script that hit a submit endpoint, saved the job id, and then counted each poll for it, thought it would be best as a Python script. I think we all have a shared sense of how long things take, and generally agree that database lookups are faster than model runs which are much faster than geoprocessing operations. If I can throw together that Python script quickly I'll include the results.